File: Server/src/main/java/com/grade/forge/faculty/entity/Faculty.java
package com.grade.forge.faculty.entity;

import com.grade.forge.coursemgmt.entity.Course;
import com.grade.forge.gradingassistant.entity.GradingAssistant;
import com.grade.forge.rubric.entity.Rubric;
import com.grade.forge.user.entity.Users;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Faculty {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String email;

    @Column(nullable = false)
    private String department;

    @Column(nullable = false)
    private String qualifications;

    private String phoneNumber;

    private String officeLocation;

    @Column(name = "is_active")
    private Boolean active;

    @CreationTimestamp
    private Instant createTime;

    @OneToMany(mappedBy = "faculty", cascade = CascadeType.ALL)
    private List<Course> courses;

    @OneToMany(mappedBy = "faculty", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Rubric> rubrics = new ArrayList<>();

    @OneToMany(mappedBy = "faculty", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<GradingAssistant> gradingAssistants = new ArrayList<>();

    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "user_id", nullable = false)
    private Users user;

}

File: Client/src/types/faculty.ts
/** Response from GET /api/v1/faculty/me */
export interface FacultyResponse {
  id?: number;
  facultyId?: number;
  userId?: number;
  name: string;
  email: string;
  department?: string;
  qualifications?: string;
  phoneNumber?: string | null;
  officeLocation?: string | null;
  officeHours?: string | null;
  role?: string;
}

/** Payload for PUT /api/v1/faculty/me */
export interface FacultyUpdateRequest {
  name?: string;
  department?: string;
  qualifications?: string;
  phoneNumber?: string | null;
  officeLocation?: string | null;
  officeHours?: string | null;
}
